### PR TITLE
Rework token component

### PR DESCRIPTION
**Pull request by pivilartisant**
Screenshot 2023-08-02 at 17 27 42

### CODE REVIEW

File: src/components/Token/Token.tsx [hunk 1]
```diff
@@ -5,30 +5,53 @@ import { ComponentPropsWithoutRef } from 'react';
...
+  const bigintBalance = BigInt(balance);
+
+  const formattedBalance = formatValue({
+    value: formatUnits(bigintBalance, decimals).toString(),
+    groupSeparator: ',',
+    decimalSeparator: '.',
+    decimalScale: 2,
+  });
+
+  // const formattedBalance = formatUnits(bigintBalance, decimals).toString();
```
Comment (mazmassa) [inline src/components/Token/Token.tsx]:
```suggestion
```

File: src/components/Token/Token.tsx [hunk 2]
```diff
@@ -5,30 +5,53 @@ import { ComponentPropsWithoutRef } from 'react';
...
+
+  const rawBalance = formatValue({
+    value: formatUnits(bigintBalance, decimals).toString(),
+    groupSeparator: ',',
+    decimalSeparator: '.',
+    decimalScale: decimals,
+  });
+
+  // const rawBalance = formatUnits(bigintBalance, decimals).toString();
```
Comment (mazmassa) [inline src/components/Token/Token.tsx]:
```suggestion
```

File: package.json [hunk 3]
```diff
@@ -20,14 +20,16 @@
...
     "copy-to-clipboard": "^3.3.3",
     "minidenticons": "^3.1.2",
     "react": "^18.2.0",
-    "react-currency-input-field": "^3.6.10",
+    "react-currency-input-field": "^3.6.11",
     "react-dom": "^18.2.0",
     "react-hot-toast": "^2.4.1",
     "react-icons": "^4.8.0",
     "tw-colors": "^1.2.5",
+    "viem": "^1.5.2",
```
Comment (mazmassa) [inline package.json:32]: I don't think we need for now. Are you not using at least...

**Review by mazmassa — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (pivilartisant) [reply package.json:32]: yes i'm using it here with the format units fn: 
```

 const formattedBalance = formatValue({
    value: formatUnits(bigintBalance, decimals).toString(),
    groupSeparator: ',',
    decimalSeparator: '.',
    decimalScale: 2,
  });
```

Comment (mazmassa) [reply package.json:32]: Ohh ok! Sorry!

**Pull request merged**